Trustroots/trustroots
# IETF language codes

#### Proposed Changes

Use the [IETF language codes](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) for i18next.

#### Motivation
Both [momentjs](http://momentjs.com/docs/#/i18n/) and [i18next](https://jsfiddle.net/jamuhl/3sL01fn0/#tabs=result) support these codes, so the pluralization and momentjs formatting works correctly.

Without this the [pluralization](https://www.i18next.com/translation-function/plurals) doesn't work.

#### Testing Instructions

* Manual: See that language switch still works. Go to a page for which a translation exists (e.g. /volunteering with Czech (česky)) and switch.

#1026 depends on this. We have pluralization there and it doesn't work currently.

* Note: The language switch labels don't matter now. Nor the picked languages. It's a design issue. We can use language codes or whatever. We'll probably list the languages which have a substantial part of the translations done.
This is how it currently looks:
![language-switch](https://user-images.githubusercontent.com/7449720/53174025-1c4d7080-35e9-11e9-91dd-0c4edfc5298e.png)


#### Challenges

API code uses [different language codes](https://github.com/Trustroots/trustroots/blob/master/config/languages/languages.json). Comparing the [original Trustroots languages](https://github.com/Trustroots/trustroots/blob/master/config/languages/languages_orig.json) and [IANA registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry), it vaguely seems that the IANA uses the shortest available code (e.g. standard `iso_639_1` rather than `iso_639_3`), whereas Trustroots prefer 3-letter codes.
_Solution:_ Since we'll have a limited number of translations for a long time or always, we can just keep the (client language list) and (its map to server language list) as needed.
_Solution:_ Alternatively we can change the server languages to fit the client needs (more, and boring, work).
